File: cpp/9_armstrong.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main(){

    int a;
    int ans = 0;
    int count = 0;

    cout<<"Enter the number : ";
    cin>>a;

    int N = a;

    while(N != 0){
        N = N/10;
        count++;
    }

    N = a;

    while(N != 0){
        ans = ans + pow((N%10),count);
        N = N/10;
    }

    (a == ans) ? cout<<"It is Armstrong" : cout<<"It is not Armstrong";
}

File: _9_armstrong.java

import java.util.Scanner;

public class _9_armstrong{
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);

        int a;
        int ans = 0;
        int count = 0;

        System.out.println("Enter the Number : ");
        a = sc.nextInt();

        int N = a;

        while(N != 0){
            N = N/10;
        count++;
        }

        N = a;

        while(N != 0){
        ans = ans + (int) Math.pow((N%10),count);
        N = N/10;
        }

        String print = (a == ans) ? "Armstrong" : "Not Armstrong";

        System.out.println(print);

    }
}